### PR TITLE
Detailed observability

### PR DESCRIPTION
**Pull request by achouippe**
This PR rename some existing observability metrics and introduce more detailed metrics: 

On message brokering:
- Rename `current_connections` to `concurrent_subscription` and split subscriptions per issuer,
- Rename `connections[created / released]` to `subscription_lifecycle[created/released]` and split per issuer,
- Add `subscription_duration_ms` to track the duration of SSE subscriptions and split per issuer,
- Rename `messages[received/published]` by `messages[published/sent]` and split per issuer (published = message published on the internal API, sent = message sent through a SSE connection),
- Add `topic_count` to track how many topics are stored in the in-memory message retention.

On system metrics:
- Add the boolean `stopping` that turned to `true` when a node is in the graceful shutdown process.

Add http metrics:
- `http_request_count[public/internal, status]` count the number of http request on each interfaces and per HTTP status, it is incremented at the end of the request processing.
- `http_request_duration_ms[public/internal]` report the duration of http request on each interfaces.

### CODE REVIEW

File: neurow/lib/neurow/broker/receiver_shard_manager.ex [hunk 1]
```diff
@@ -35,20 +35,26 @@ defmodule Neurow.Broker.ReceiverShardManager do
...
-    Stats.inc_history_rotate()
+    Neurow.Stats.MessageBroker.inc_history_rotate()
 
-    all_pids(fn {_, pid} ->
+    Enum.each(receiver_shards(), fn {_shard, pid} ->
       send(pid, {:rotate})
     end)
   end
 
+  def topic_count do
```
Comment (papey) [inline neurow/lib/neurow/broker/receiver_shard_manager.ex:52]: **Suggestion** 🤔 : if we prefer the pipeline route, i would have write it like this

```
  receiver_shards()
  |> Enum.map(&Neurow.Broker.ReceiverShard.topic_count(&1.pid))
  |> Enum.sum()
```

File: neurow/lib/neurow/broker/receiver_shard.ex [hunk 2]
```diff
@@ -73,4 +77,9 @@ defmodule Neurow.Broker.ReceiverShard do
     :ets.delete_all_objects(table_1)
     {:reply, :ok, {table_0, table_1}}
   end
+
+  @impl true
+  def handle_call({:topic_count}, _from, {table_0, table_1}) do
```
Comment (papey) [inline neurow/lib/neurow/broker/receiver_shard.ex:82]: **Question** ❓ : how is this consumed / aggregate in a multi nodes setup ?

File: neurow/lib/neurow/public_api/endpoint.ex [hunk 3]
```diff
@@ -244,10 +247,26 @@ defmodule Neurow.PublicApi.Endpoint do
 
           # Consume useless messages to avoid memory overflow
           _ ->
-            conn |> loop(sse_timeout_ms, keep_alive_ms, last_message_ms, last_ping_ms, jwt_exp_s)
+            conn
```
Comment (papey) [inline neurow/lib/neurow/public_api/endpoint.ex:250]: **Suggestion** 🤔 : i we do one pipe, i think it's clearer to have just put the param directly in the func call

**Review by papey — APPROVED** [review]
LGTM overall, some syntax suggestions

File: neurow/lib/neurow/broker/receiver_shard.ex (hunk 2 above)
Comment (achouippe) [reply neurow/lib/neurow/broker/receiver_shard.ex:82]: Each node has its own copy of all topics. Once aggregated in a single dashboard, we will be able to check if all nodes have actually the same topic count.

**Pull request merged**